docs/general/sentence_processing_and_feature_extraction_techniques_for_luigi.md

```diff
@@ -0,0 +1,109 @@
+| About Roboy                       | `nlp.prop_roboy`              | Detecting whether sentence is about Roboy                   | 'you', 'roboy', 'robot', 'roboboy', ... |
+| Yes-No                            | `nlp.prop_yesno`              | Detecting answers to yes-no questions                       | Checking for 'yes', 'no', 'i don't know', 'probably', 'probably not' and synonyms of these                           |
+| Sentence Type: Question           | `nlp.sig_is_question`         | Emitted if the input sentence is a question                 |                                                |
+| Play Game                         | `nlp.sig_intent_play`         | Emitted when the interlocutor wants to play a game          | input: "I want to play", "I like games" or something similar    |
+
+### Using the Features
+
+#### Detailed POS Tagging
+
+This feature can be used to understand the requested flavors or payment options. 
+Roboy's flavor and payment options are predefined so any match between our database and incoming token
+can help us to identify the requested options. Such as:
+
+- vanilla
+- chocolate
+- lemon
+- strawberry
+
+or 
+
+- credit card
+- cash
+- bitcoin
+
+Alternatively by using POS, Roboy can understand whether the sentence is a request or not by only parsing the verb.
+For example if the customer uses one of the following phrases, it is easy to detect that customer has a request:
+
+- would like
+- can have
+- can get
+- will have
+- choose
+- wish
+- desire
+- prefer
+- could do with
+- have an urge for
+
+#### Yes-No Property
+
+This feature can be used to understand a given question's positive/negative answer to proceed such as:
+
+- Do you want the usual?
+- Do you want ice cream?
+
+Example: State that reads the "yesno" property
+
+```python
+import ravestate as rs
+import ravestate_nlp as nlp
+import ravestate_rawio as rawio
+
+@rs.state(
+    cond=nlp.prop_yesno.changed(),  # state reacts to a change in the 'yesno' property
+    read=nlp.prop_yesno,  # state is allowed to read the 'yesno' property
+    write=rawio.prop_out)  # state is allowed to write to the output property
+def customer_wants_ice_cream(ctx: ContextWrapper):
+    if ctx[nlp.prop_yesno] == "yes":
+        ctx[rawio.prop_out] = "Your ice cream is coming right up!"
+
+```
+
+#### Lemmatization
+
+After tokenizing the sentences, roboy needs to match the strings in order to understand the context.
+Lemmatization allows to get the base form of words which is the required form for matching.
+
+
+#### Named Entity Recognition (NER)
+
+NER can be used in order to understand the quantity of ice cream, payment or any matter that requires specifics.
+The following figure includes the most possible types of use, please check 
+https://spacy.io/api/annotation#named-entities for further types.
+
+| Type                              | DESCRIPTION
```

Comment:
Please insert a reference to spacy's interactive NER detector here: https://explosion.ai/demos/displacy-ent

You will notice that it is not very reliable...